From ae7bc20fced432d38b48cc9960a5f616bf32661c Mon Sep 17 00:00:00 2001
From: Karthikeyan Periyasamy <quic_periyasa@quicinc.com>
Date: Tue, 2 Aug 2022 20:08:36 +0530
Subject: [PATCH v2] wifi: ath12k: fix lock/unlock issue in MLO offset event
 handler

In the MLO offset event handler, ab->base_lock get unlocked
instead of ar->data_lock. This leads to hang/deadlock in the
system. So fix this issue by unlock with proper radio lock.

Found in code review. This code change is applicable for all
the platform.

Tested-on: QCN9274 hw1.0 PCI WLAN.HK.2.8.0.1.r1-00008-QCAHKSWPL_SILICONZ-1-private

Signed-off-by: Karthikeyan Periyasamy <quic_periyasa@quicinc.com>
---
 v2:
   - fixed spell in the commit message

 drivers/net/wireless/ath/ath12k/dp_rx.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/net/wireless/ath/ath12k/dp_rx.c
+++ b/drivers/net/wireless/ath/ath12k/dp_rx.c
@@ -1819,7 +1819,7 @@ static void ath12k_htt_mlo_offset_event_
 	pdev->timestamp.mlo_comp_clks = __le32_to_cpu(msg->mlo_comp_clks);
 	pdev->timestamp.mlo_comp_timer = __le32_to_cpu(msg->mlo_comp_timer);
 
-	spin_unlock_bh(&ab->base_lock);
+	spin_unlock_bh(&ar->data_lock);
 }
 
 void ath12k_htt_vdev_txrx_stats_handler(struct ath12k_base *ab,
